pkg/apis/core/types.go

```diff
@@ -3486,6 +3486,10 @@ type ServiceStatus struct {
 	// if one is present.
 	// +optional
 	LoadBalancer LoadBalancerStatus
+
+	//Current service state
+	// +optional
+	Conditions []ServiceCondition
```

Comment:
The KEP says: "Our feature does not introduce new values or new fields"

 am not saying that adding a field is wrong, but it is not part of the committed KEP.  I re-read the discussion on the KEP and we agreed to this in theory, but it was never updated

Reply:
We should use `metav1.Condition` from staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

Reply:
> but it is not part of the committed KEP. I re-read the discussion on the KEP and we agreed to this in theory, but it was never updated

Hmmm, we have in the KEP the following: 
```
In order to provide a way for the CPI to indicate port statuses to the user we would add the following new portStatus list of port statuses to Service.status.loadBalancer.ingress, so the CPI can indicate the status of the LB. Also we add a conditions list of conditions to the Service.status.loadBalancer, with the first official condition LoadBalancerMixedProtocolNotSupported defined.
```

And then there is the specification of these fields, too.

Am I missing something?

Reply:
So, should I remove this API change (ServiceStatus) from this PR? I put it in here, as it is part of the KEP.

Reply:
You are right, I guess I skimmed too fast.  Sorry.

Reply:
> So, I started using metav1.Condition there instead of ServiceConditions, but as I can see, bazel-test fails due to that https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/94028/pull-kubernetes-bazel-test/1296919306033958912

chatted in slack, but for posterity, looks like metav1.Condition types should be added to https://github.com/kubernetes/kubernetes/blob/c3b888f647f99eba246468ac3e13afc49542c06d/pkg/controlplane/import_known_versions_test.go#L53-L73

Reply:
@liggitt Thank you for your help. Indeed, it is OK now.